mlflow/gateway/config.py

```diff
@@ -159,16 +159,13 @@ def _validate_field_compatibility(cls, info: dict[str, Any]):
         return info
 
     if IS_PYDANTIC_V2_OR_NEWER:
-        from pydantic import model_validator as _model_validator
 
-        @_model_validator(mode="before")
+        @model_validator(mode="after")
         def validate_field_compatibility(cls, info: dict[str, Any]):
             return cls._validate_field_compatibility(info)
-
     else:
-        from pydantic import root_validator as _root_validator
 
-        @_root_validator(pre=False)
+        @model_validator(mode="after")
         def validate_field_compatibility(cls, config: dict[str, Any]):
             return cls._validate_field_compatibility(config)
```

Comment:
I think this if can now be deleted from here no? I mean it is exactly the same code